src/gcs.rs

```diff
@@ -640,23 +642,64 @@ mod tests {
     async fn get_private_object() {
         init();
 
+        // define the private object
         let private_bucket = test_bucket();
         let filename = "get_private_object.txt";
 
+        // generate test data of a few kB
+        let file_length = 4096;
+        let rand_string: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(file_length)
+            .map(char::from)
+            .collect();
+        // upload the test data
+        let client = &new_client();
+        let mut object_cursor = create_object_with_client(client, &private_bucket, filename)
+            .await
+            .unwrap();
+        append_bytes_with_client(
+            client,
+            &mut object_cursor,
+            &Vec::from(rand_string.into_bytes()),
+        )
+        .await
+        .unwrap();
+        finalize_upload_with_client(client, &mut object_cursor)
+            .await
+            .unwrap();
+
+        // test getting the private object
         let url = format!(
             "https://www.googleapis.com/storage/v1/b/{}/o/{}",
             private_bucket, filename
         );
 
+        // get object metadata
         let object_url = Url::parse(&url).unwrap();
-        let object: Object = get_object(object_url).await.unwrap();
+        let object = get_object(object_url).await.unwrap();
         println!("Object has {} bytes", object.size);
+        assert!(object.size == file_length as u64);
+        assert!(object.name == filename);
 
-        let bytes: Vec<u8> = get_bytes(&object, 0, 769).await.unwrap();
+        // get whole object
+        let bytes: Vec<u8> = get_bytes(&object, 0, file_length as u64).await.unwrap();
+        let bytes_len = bytes.len();
         println!("Got back:\n {}", String::from_utf8(bytes).unwrap());
+        assert!(bytes_len == file_length);
 
-        let offset_bytes: Vec<u8> = get_bytes(&object, 6, 769).await.unwrap();
-        println!("Got back:\n {}", String::from_utf8(offset_bytes).unwrap());
+        // get partial object
+        let start_offset = 6;
+        let offset_bytes: Vec<u8> = get_bytes(&object, start_offset as u64, file_length as u64)
+            .await
+            .unwrap();
+        let offset_bytes_len = offset_bytes.len();
+        println!(
+            "Asked for file starting at byte {}. Got back:\n {}",
+            start_offset,
+            String::from_utf8(offset_bytes).unwrap()
```

Comment:
As above, make sure that the slice or rand_string matches this offset (to test that we got the same bytes back)

Reply:
Done.